File: app/src/main/java/com/softwareverde/tidyduck/most/RecordField.java
package com.softwareverde.tidyduck.most;

public class RecordField {
    private Long _id;
    private String _fieldName;
    private String _fieldIndex;
    private String _fieldDescription;
    private MostType _fieldType;

    public Long getId() {
        return _id;
    }

    public void setId(Long id) {
        _id = id;
    }

    public String getFieldName() {
        return _fieldName;
    }

    public void setFieldName(String fieldName) {
        _fieldName = fieldName;
    }

    public String getFieldIndex() {
        return _fieldIndex;
    }

    public void setFieldIndex(String fieldIndex) {
        _fieldIndex = fieldIndex;
    }

    public String getFieldDescription() {
        return _fieldDescription;
    }

    public void setFieldDescription(String fieldDescription) {
        _fieldDescription = fieldDescription;
    }

    public MostType getFieldType() {
        return _fieldType;
    }

    public void setFieldType(MostType fieldType) {
        _fieldType = fieldType;
    }
}

File: www/js/classes/record-field.js
class RecordField {
    static fromJson(json) {
        if (json == null) {
            return null;
        }

        const fieldType = MostType.fromJson(json.fieldType);

        const recordField = new RecordField();

        recordField.setId(json.id);
        recordField.setFieldName(json.fieldName);
        recordField.setFieldIndex(json.fieldIndex);
        recordField.setFieldDescription(json.fieldDescription);
        recordField.setFieldType(fieldType);

        return recordField;
    }

    static toJson(recordField) {
        if (recordField == null) {
            return null;
        }
        return {
            id:                 recordField.getId(),
            fieldName:          recordField.getFieldName(),
            fieldIndex:         recordField.getFieldIndex(),
            fieldDescription:   recordField.getFieldDescription(),
            fieldType:          MostType.toJson(recordField.getFieldType())
        };
    }

    constructor() {
        this._id                = null;
        this._fieldName         = null;
        this._fieldIndex        = null;
        this._fieldDescription  = null;
        this._fieldType         = null;
    }


    setId(id) {
        this._id = id;
    }

    getId() {
        return this._id;
    }

    setFieldName(fieldName) {
        this._fieldName = fieldName;
    }

    getFieldName() {
        return this._fieldName;
    }

    setFieldIndex(fieldIndex) {
        this._fieldIndex = fieldIndex;
    }

    getFieldIndex() {
        return this._fieldIndex;
    }

    setFieldDescription(fieldDescription) {
        this._fieldDescription = fieldDescription;
    }

    getFieldDescription() {
        return this._fieldDescription;
    }

    setFieldType(fieldType) {
        this._fieldType = fieldType;
    }

    getFieldType() {
        return this._fieldType;
    }

}